src/components/Sidebar/BranchTransition.tsx

```diff
@@ -1,22 +1,75 @@
-import React, { FC, memo } from 'react';
+import React, { FC, memo, useState, useCallback, ChangeEvent } from 'react';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faPlus, faTools } from '@fortawesome/free-solid-svg-icons';
-
+import DropDown from 'components/elements/DropDown';
+import { TextField, FormControl } from '@material-ui/core';
+import { setTransitionCondition, setTransitionConditionDescription } from 'utils/builder';
 import { SidebarHeader, SidebarButton } from '.';
 import { Pathway, Transition } from 'pathways-model';
+import { useCriteriaContext } from 'components/CriteriaProvider';
 import useStyles from './styles';
 
 interface BranchTransitionProps {
   pathway: Pathway;
+  currentNodeKey: string;
   transition: Transition;
   updatePathway: (pathway: Pathway) => void;
 }
 
-const BranchTransition: FC<BranchTransitionProps> = ({ pathway, transition, updatePathway }) => {
+const BranchTransition: FC<BranchTransitionProps> = ({
+  pathway,
+  currentNodeKey,
+  transition,
+  updatePathway
+}) => {
+  const { criteria } = useCriteriaContext();
+  const criteriaOptions = criteria.map(c => ({ value: c.label, label: c.label }));
```

Comment:
So the label isn't necessarily unique. We should use the id as the value. Then line 43 can be updated to search for the id
```suggestion
  const criteriaOptions = criteria.map(c => ({ value: c.id, label: c.label }));
```